common-utils/src/common/deployment/__init__.py

```diff
@@ -318,33 +300,73 @@ def push_deployment(
             infrastructure (str): ECS Task block name to use for the flow execution environment.
             flow_path (Path): Relative path to the flow python file.
             flow_function_name: (str): Name of the flow function to deploy.
-            skip_upload (bool): Whether to skip upload. Will be set to True after first deployment pushed for flow.
+            flow_name: (str): Properly enforced flow name for deployment.
+
         """
-        project_name = PYPROJECT_METADATA.project_name
-        deployment_name = standard_slugify(self.deployment_name)
+        # start building the cli arguments
+        deployment_name = f"{standard_slugify(self.deployment_name)}-{DEPLOYMENT_TYPE}"
         flow_file_name = flow_path.name
-        overrides = f"--override cpu={self.cpu} --override memory={self.memory}"
+        task_customizations = [
+            {
+                "op": "add",
+                "path": "/overrides/cpu",
+                "value": self.cpu,
+            },
+            {
+                "op": "add",
+                "path": "/overrides/memory",
+                "value": self.memory,
+            },
+            {
+                "op": "add",
+                "path": "/networkConfiguration/awsvpcConfiguration/subnets",
+                "value": json.loads(AWS_SUBNETS),
+            },
+            {
+                "op": "add",
+                "path": "/networkConfiguration/awsvpcConfiguration/securityGroups",
+                "value": json.loads(AWS_SECURITY_GROUPS),
+            },
+            {
+                "op": "add",
+                "path": "/networkConfiguration/awsvpcConfiguration/assignPublicIp",
+                "value": "DISABLED",
+            },
+        ]
+        env_overrides = " ".join(
+            [
+                f"--override env.{i.dict()['envar_name']}='{i.dict()['envar_value']}'"
+                for i in self.envars
+            ]
+        )
+        task_overrides = (
+            f"""--override task_customizations='{json.dumps(task_customizations)}'"""
+        )
+        overrides = env_overrides + " " + task_overrides
         schedule = self._get_schedule_arg()
         params = f"'{json.dumps(self.parameters)}'"
-        skip_upload_flag = ""
-        if skip_upload:
-            skip_upload_flag = "--skip-upload"
-        # We are using the escape hatch envar PREFECT_PYPROJECT_PATH because of limitation in prefect CLI
-        # We may remove this in the future.
+        # envars here are used for local testing purposes
+        infra_arg = os.getenv(
+            "POCKET_PREFECT_INFRASTRUCTURE_BLOCK", f"-ib ecs-task/{infrastructure}"
+        )
+        github_block = os.getenv(
+            "POCKET_PREFECT_GITHUB_BLOCK", f"data-flows-{DEPLOYMENT_TYPE}"
+        )
+        # run deployment cli using helper function
```

Comment:
we want to be able to override infrastructure and storage to enable deployment and execution on local Prefect services